src/middlewared/middlewared/alert/source/volume_status.py

```diff
@@ -81,3 +96,14 @@ async def enabled(self):
             return status in ("MASTER", "SINGLE")
 
         return True
+
+    def pools_states(self):
```

Comment:
We already have a function that does this. It's in `utils/zfs.py` named `query_fast_imported_impl`